screens/MapScreen.tsx

```diff
@@ -302,15 +306,20 @@ export default function MapScreen() {
     setFilteredEvents([]);
     setIsFiltering(false);
     setIsFilterModalVisible(false);
-    setSelectedDate(new Date());
+    setSelectedStartDate(new Date());
+    setSelectedEndDate(new Date());
   };
 
   const onFilterDateEvents: () => void = () => {
     analytics().logEvent("custom_log", {
       description: "--- Analytics: screens -> MapScreen -> onFilterDateEvents",
     });
     const filtered: Event[] = events.filter((event: Event) => {
-      return new Date(event.date).toLocaleDateString() === selectedDate.toLocaleDateString()
+      const eventDate = new Date(event.date).getTime();
+      const startDate = selectedStartDate ? selectedStartDate.getTime() : 0;
+      const endDate = selectedEndDate ? selectedEndDate.getTime() : Number.MAX_SAFE_INTEGER;
```

Comment:
Why Number.MAX_SAFE_INTEGER?

Reply:
Because MAX_SAFE_INTEGER is largest positive number so any date timestamp not greater than the MAX_SAFE_INTEGER  so use it.